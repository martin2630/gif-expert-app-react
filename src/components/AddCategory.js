import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    // const [{setCategories}] = this.props;
    const initialInputValue = ''
    const [inputValue, setInputValue] = useState(initialInputValue);

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
        console.log(" handleInputChange llamado");
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length>2){
            setCategories(cats => [inputValue, ...cats])
            setInputValue(initialInputValue)
        }
        return 
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                onChange={handleInputChange} 
                value={inputValue} 
                placeholder="Rick y Morty"
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
