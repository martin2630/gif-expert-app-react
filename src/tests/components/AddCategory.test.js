import { React } from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe("Pruebas en <AddCategory />", () => {
    const setCateries = () => {
        console.log("setCategories");
    }
    const wrapper = shallow(<AddCategory setCategories={setCateries} />)
    

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de validar que el input cambio', () => {
        const input = wrapper.find('input');
        const value = 'Holi';
        
        input.simulate("change", { target: { value } })
        expect(wrapper.find('p').text()).toBe(value)
    })
    
})