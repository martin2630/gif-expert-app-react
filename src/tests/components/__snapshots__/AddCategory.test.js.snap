// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Rick y Morty"
    type="text"
    value=""
  />
</form>
`;
